.github/workflows/tests.yml

```diff
@@ -21,129 +21,77 @@ jobs:
+      - name: Building...
+        shell: pwsh
+        run: dotnet build 
+        working-directory: src
+      - name: Installing Browsers and dependencies...
+        shell: pwsh
         run: |
-          dotnet dev-certs https --clean
-          dotnet dev-certs https -ep src/Playwright.Tests.TestServer/testCert.cer
-      - name: Clean
-        run: dotnet clean ./src/Playwright.sln --configuration Debug && dotnet nuget locals all --clear
-      - name: Download Drivers
-        run: dotnet run -p ./src/tools/Playwright.Tooling/Playwright.Tooling.csproj -- download-drivers --basepath .
-      - name: Download Browsers
-        run: dotnet run -p ./src/Playwright -f net5.0 install
-      - name: Run tests (!Linux)
+          dotnet run -p .\src\Playwright\Playwright.csproj -f net5.0 -- install
+          dotnet run -p .\src\Playwright\Playwright.csproj -f net5.0 -- install-deps
+      - name: Running tests...
         if: ${{ matrix.os != 'ubuntu-latest' }}
+        shell: pwsh
         env:
           BROWSER: ${{ matrix.product }}
-        run: |
-          dotnet test ./src/Playwright.Tests/Playwright.Tests.csproj -c Debug -f net5.0 --logger "trx;LogFileName=TestResults.xml" -- NUnit.NumberOfTestWorkers=1
-      - name: Run tests (Linux)
+        run: 
+          dotnet test ./Playwright.Tests/Playwright.Tests.csproj -c Debug -f net5.0 --logger "trx;LogFileName=TestResults.xml" -- NUnit.NumberOfTestWorkers=1
+        working-directory: src
+      - name: Running tests... (Linux)
         if: ${{ matrix.os == 'ubuntu-latest' }}
+        shell: bash
         env:
           BROWSER: ${{ matrix.product }}
+        working-directory: src
         run: |
-          xvfb-run --auto-servernum --server-args="-screen 0 1280x960x24" -- bash -c "ulimit -c unlimited && dotnet test ./src/Playwright.Tests/Playwright.Tests.csproj -c Debug -f net5.0 --logger \"trx\" -- NUnit.NumberOfTestWorkers=1"
+          xvfb-run --auto-servernum --server-args="-screen 0 1280x960x24" -- bash -c "ulimit -c unlimited && dotnet test ./Playwright.Tests/Playwright.Tests.csproj -c Debug -f net5.0 --logger \"trx\" -- NUnit.NumberOfTestWorkers=1"
+
   test_net31:
-    name: Chromium on Windows using .NET 3.1
-    runs-on: windows-latest
+    name: Chromium on Ubuntu using .NET 3.1
+    runs-on: ubuntu-latest
     steps:
-      - name: Install Media Pack
-        if: ${{ matrix.os == 'windows-latest' }}
-        shell: powershell
-        run: Install-WindowsFeature Server-Media-Foundation
       - uses: actions/checkout@v2
-      - uses: microsoft/playwright-github-action@v1
-      - name: Setup .NET Core (On Mac)
-        if: ${{ matrix.os == 'macos-latest' }}
-        uses: actions/setup-dotnet@v1
         with:
-          dotnet-version: 3.1.100
-        env:
-          DOTNET_INSTALL_DIR: /users/runner/alternate/dotnet
-      - name: Setup .NET Core (Not on mac)
-        if: ${{ matrix.os != 'macos-latest' }}
+          submodules: true
+          fetch-depth: 0
```

Comment:
why fetch-depth: 0?

Reply:
IIRC this was needed to ensure I get the submodule at the right commit

Reply:
Also, that was true for the integrity check PR. It's possible I could get away with `1` here. I'll give it a go. It might speed it up ever so slightly.